cmd/reminder/app/root.go

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package app provides the cli subcommands for managing the reminder service
+package app
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/stacklok/minder/internal/config"
+	reminderconfig "github.com/stacklok/minder/internal/config/reminder"
+	"github.com/stacklok/minder/internal/util/cli"
+)
+
+var (
+	// RootCmd represents the base command when called without any subcommands
+	RootCmd = &cobra.Command{
+		Use:   "reminder",
+		Short: "reminder controls the reminder service",
+		Long:  `reminder sends entity reconciliation requests to the minder server`,
+	}
+)
+
+const configFileName = "reminder-config.yaml"
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+func Execute() {
+	RootCmd.SetOut(os.Stdout)
+	RootCmd.SetErr(os.Stderr)
+	err := RootCmd.ExecuteContext(context.Background())
+	cli.ExitNicelyOnError(err, "Error executing root command")
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+	reminderconfig.SetViperDefaults(viper.GetViper())
+
+	if err := reminderconfig.RegisterReminderFlags(viper.GetViper(), RootCmd.PersistentFlags()); err != nil {
+		RootCmd.PrintErrln("Error registering reminder flags")
+		os.Exit(1)
+	}
+
+	RootCmd.PersistentFlags().String("config", "", fmt.Sprintf("config file (default is $PWD/%s)", configFileName))
+	if err := viper.BindPFlag("config", RootCmd.PersistentFlags().Lookup("config")); err != nil {
+		RootCmd.Printf("error: %s", err)
+		os.Exit(1)
+	}
+}
+
+func initConfig() {
+	cfgFile := viper.GetString("config")
+	cfgFileData, err := config.GetConfigFileData(cfgFile, filepath.Join(".", configFileName))
+	if err != nil {
+		RootCmd.PrintErrln(err)
+		os.Exit(1)
+	}
+
+	keysWithNullValue := config.GetKeysWithNullValueFromYAML(cfgFileData, "")
+	if len(keysWithNullValue) > 0 {
+		RootCmd.PrintErrln("Error: The following configuration keys are missing values:")
+		for _, key := range keysWithNullValue {
+			RootCmd.PrintErrln("Null Value at: " + key)
+		}
+		os.Exit(1)
+	}
+
+	err = reminderconfig.ValidateConfig(cfgFileData)
```

Comment:
> modulo the above "null keys when we need a value" check

The null values check solves a different purpose. It is there to prevent corruption of viper key value store. It is there in all other cli's too. 

> can we consolidate one into the other?

I would rather not. The validation portion has a different method signature than the null value one, and doing something to workaround would break the pattern across CLIs rather than giving any form of uniformity.

> Ideally, the code for initConfig would look similar for minder, server and reminder...)

It is almost the same, except reminder does validation.